src/lib.rs

```diff
@@ -101,12 +124,38 @@ impl<T> Gc<T> {
 
             AllocMetadata::insert(objptr as usize, layout.size(), true);
             let headerptr = objptr.sub(size_of::<usize>());
-            ptr::write(headerptr as *mut usize, 1);
+            ptr::write(headerptr as *mut GcHeader, GcHeader::new());
             objptr as *mut T
         }
     }
 }
 
+impl GcHeader {
+    pub(crate) fn new() -> Self {
+        Self(0)
+    }
+
+    pub(crate) fn mark_bit(&self) -> bool {
+        ((self.0 >> 0) & 1) == 1
+    }
+
+    pub(crate) fn set_mark_bit(&mut self, mark: bool) {
+        if mark {
+            self.0 |= 1 << 0
+        } else {
+            self.0 &= !(1 << 0)
```

Comment:
I think it was naughty of Rust to use `!` for bitwise compliment on integers instead of `~`.

Reply:
I agree